### PR TITLE
chore: migrate to bloc 8.0.0, flow_builder 0.0.6 and mocktail 0.2.0

### PR DESCRIPTION
**Pull request by yeikel16**
Update dependencies :
- `bloc` to 8.0.1
- `flow_builder` to 0.0.6
- `flutter_bloc` to 8.0.0
- `bloc_test` to 9.0.1
- `mocktail` to 0.2.0

No longer implicitly requires `registerFallbackValue` for events and states

### CODE REVIEW

File: examples/form_flow/packages/formz_inputs/lib/src/pin.dart [hunk 1]
```diff
@@ -21,10 +21,13 @@ class PinFormInput extends FormzInput<String, PinValidationError> {
...
-      assert(() {
-        _maxLength = 4;
-        return true;
-      }());
+      assert(
+        () {
+          _maxLength = 4;
+          return true;
+        }(),
+        'Maximum length the pin requires to be valid.',
```
Comment (marcossevilla) [inline examples/form_flow/packages/formz_inputs/lib/src/pin.dart:29]:
```suggestion
        'Pin’s maximum length defaults to 4.',
```

File: examples/launch_analytics/lib/main.dart [hunk 2]
```diff
@@ -8,17 +10,39 @@ import 'package:path_provider/path_provider.dart';
 
 Future<void> main() async {
```
Comment (marcossevilla) [inline examples/launch_analytics/lib/main.dart:11]: can we use a bootstrap function here too?

File: examples/launch_analytics/test/app/bloc/app_bloc_test.dart [hunk 3]
```diff
@@ -14,134 +14,156 @@ class MockAnalyticsRepository extends Mock implements AnalyticsRepository {}
...
+            when(() => analyticsRepository.getOpeningsCount()).thenReturn(0);
+
+            final bloc = AppBloc(localAnalyticsRepository: analyticsRepository);
+
+            expect(bloc.state, AppInitial());
+
+            // ignore: avoid_single_cascade_in_expression_statements
+            bloc..add(AppAnalyticsChecked());
+
+            await expectLater(
```
Comment (marcossevilla) [inline examples/launch_analytics/test/app/bloc/app_bloc_test.dart:161]: we should revert these tests to use bloc_test before merging

File: examples/preference_navigation/analysis_options.yaml [hunk 4]
```diff
@@ -1 +1,6 @@
-include: package:very_good_analysis/analysis_options.yaml
+include: package:very_good_analysis/analysis_options.2.4.0.yaml
+linter:
+  rules:
+    require_trailing_commas: false
```
Comment (marcossevilla) [inline examples/preference_navigation/analysis_options.yaml:4]: why are we ignoring these?
```suggestion
```

File: examples/preference_navigation/packages/preferences/hive_preferences_repository/analysis_options.yaml [hunk 5]
```diff
@@ -1 +1,4 @@
-include: package:very_good_analysis/analysis_options.2.3.0.yaml
\ No newline at end of file
+include: package:very_good_analysis/analysis_options.2.4.0.yaml
+linter:
+  rules:
+    require_trailing_commas: false
```
Comment (marcossevilla) [inline examples/preference_navigation/packages/preferences/hive_preferences_repository/analysis_options.yaml:4]:
```suggestion

```

File: examples/preference_navigation/analysis_options.yaml [hunk 6]
```diff
@@ -1 +1,6 @@
...
+    library_private_types_in_public_api: false
```
Comment (marcossevilla) [inline examples/preference_navigation/analysis_options.yaml:5]:
```suggestion

```

File: examples/preference_navigation/packages/preferences/shared_preferences_repository/analysis_options.yaml [hunk 7]
```diff
@@ -1 +1,4 @@
-include: package:very_good_analysis/analysis_options.2.3.0.yaml
\ No newline at end of file
+include: package:very_good_analysis/analysis_options.2.4.0.yaml
+linter:
+  rules:
+    require_trailing_commas: false
```
Comment (marcossevilla) [inline examples/preference_navigation/packages/preferences/shared_preferences_repository/analysis_options.yaml:4]:
```suggestion

```

File: examples/profile_accounts/analysis_options.yaml [hunk 8]
```diff
@@ -1,4 +1,5 @@
-include: package:very_good_analysis/analysis_options.yaml
+include: package:very_good_analysis/analysis_options.2.4.0.yaml
 linter:
   rules:
-    public_member_api_docs: false
\ No newline at end of file
+    public_member_api_docs: false
+    library_private_types_in_public_api: false
```
Comment (marcossevilla) [inline examples/profile_accounts/analysis_options.yaml:5]:
```suggestion

```

**Review by marcossevilla — CHANGES_REQUESTED** [review]
TAFN with lints and we should keep it as a draft until we get the bloc tests right

File: examples/preference_navigation/packages/preferences/hive_preferences_repository/analysis_options.yaml [hunk 9]
```diff
@@ -1 +1,3 @@
-include: package:very_good_analysis/analysis_options.2.3.0.yaml
\ No newline at end of file
+include: package:very_good_analysis/analysis_options.2.4.0.yaml
+linter:
```
Comment (marcossevilla) [inline examples/preference_navigation/packages/preferences/hive_preferences_repository/analysis_options.yaml:2]:
```suggestion

```

File: examples/preference_navigation/packages/preferences/hive_preferences_repository/analysis_options.yaml [hunk 10]
```diff
@@ -1 +1,3 @@
...
+  rules:
```
Comment (marcossevilla) [inline examples/preference_navigation/packages/preferences/hive_preferences_repository/analysis_options.yaml:3]:
```suggestion

```

File: examples/preference_navigation/packages/preferences/shared_preferences_repository/analysis_options.yaml [hunk 11]
```diff
@@ -1 +1,3 @@
-include: package:very_good_analysis/analysis_options.2.3.0.yaml
\ No newline at end of file
+include: package:very_good_analysis/analysis_options.2.4.0.yaml
+linter:
+  rules:
```
Comment (marcossevilla) [inline examples/preference_navigation/packages/preferences/shared_preferences_repository/analysis_options.yaml:3]:
```suggestion

```

File: examples/preference_navigation/packages/preferences/shared_preferences_repository/analysis_options.yaml [hunk 12]
```diff
@@ -1 +1,3 @@
-include: package:very_good_analysis/analysis_options.2.3.0.yaml
\ No newline at end of file
+include: package:very_good_analysis/analysis_options.2.4.0.yaml
+linter:
```
Comment (marcossevilla) [inline examples/preference_navigation/packages/preferences/shared_preferences_repository/analysis_options.yaml:2]:
```suggestion

```

File: examples/preference_navigation/pubspec.yaml [hunk 13]
```diff
@@ -7,12 +7,14 @@ environment:
   sdk: ">=2.12.0 <3.0.0"
 
 dependencies:
-  bloc: ^7.2.0-dev.3
+  bloc: ^8.0.0
```
Comment (marcossevilla) [inline examples/preference_navigation/pubspec.yaml:10]:
```suggestion
  bloc: ^8.0.1
```

File: examples/preference_navigation/pubspec.yaml [hunk 14]
```diff
@@ -24,12 +26,12 @@ dependencies:
     path: packages/preferences/shared_preferences_repository
 
 dev_dependencies:
-  bloc_test: ^8.1.0
+  bloc_test: ^9.0.0
```
Comment (marcossevilla) [inline examples/preference_navigation/pubspec.yaml:29]:
```suggestion
  bloc_test: ^9.0.1
```

File: examples/launch_analytics/test/app/bloc/app_bloc_test.dart [hunk 15]
```diff
@@ -14,134 +14,164 @@ class MockAnalyticsRepository extends Mock implements AnalyticsRepository {}
...
+            verify<dynamic>(
+              () => storage.write('AppBloc', any<Map<String, dynamic>>()),
+            ).called(1);
+          },
+          storage: storage,
+        );
+      },
+    );
 
     group('AppAnalyticsChecked', () {
```
Comment (marcossevilla) [inline examples/launch_analytics/test/app/bloc/app_bloc_test.dart:112]: can we revert these to use bloc_test?

File: examples/launch_analytics/test/app/bloc/app_bloc_test.dart [hunk 16]
```diff
@@ -14,134 +14,164 @@ class MockAnalyticsRepository extends Mock implements AnalyticsRepository {}
...
-
-        verify<dynamic>(() => storage.read('AppBloc')).called(1);
-        verify<dynamic>(
-          () => storage.write('AppBloc', any<Map<String, dynamic>>()),
-        ).called(2);
-      });
+        'emits [AppAnalyticsLoaded] and increases and read openings '
+        'when AppAnalyticsChecked is added',
+        () {
+          HydratedBlocOverrides.runZoned(
```
Comment (marcossevilla) [inline examples/launch_analytics/test/app/bloc/app_bloc_test.dart:117]: can we setup this override on the setUp method and just stub its dependencies on every test?

**Review by marcossevilla — CHANGES_REQUESTED** [review]
TAFN, mostly reverting back the bloc tests, updating some versions to latest and some nits on bloc tests for BlocOverrides 👍

File: examples/launch_analytics/test/app/bloc/app_bloc_test.dart (hunk 15 above)
Comment (yeikel16) [reply examples/launch_analytics/test/app/bloc/app_bloc_test.dart:112]: When I use blocTest I have no way of knowing the first state when creating the AppBloc instance. Any suggestion?

Comment (marcossevilla) [reply examples/launch_analytics/test/app/bloc/app_bloc_test.dart:118]: if we already have a test that verifies the initial state, we can remove the initial state from the list of expected emitted states

**Review by marcossevilla — APPROVED** [review]

**Pull request merged**